projects/kit/utils/math/key-steps.ts

```diff
@@ -33,11 +33,19 @@ export function tuiKeyStepValueToPercentage(
     value: number,
     keySteps: TuiKeySteps,
 ): number {
+    const stepsBoundariesByFn = tuiFindKeyStepsBoundariesByFn(
+        keySteps,
+        ([_, keyStepValue]) => value <= keyStepValue,
+    );
+
+    const [lower, upper] = stepsBoundariesByFn;
+
+    if (!(lower || upper)) {
```

Comment:
I am not sure that we need to change anything inside this file.

Reply:
I got it, thanks!

Let's change `tuiFindKeyStepsBoundariesByFn` function instead:


```ts
function tuiFindKeyStepsBoundariesByFn(
    keySteps: TuiKeySteps,
    fn: ([keyStepPercent, keyStepValue]: [number, number]) => boolean,
): [[number, number], [number, number]] {
    const keyStepUpperIndex = keySteps.findIndex((keyStep, i) => i && fn(keyStep));

    const lowerStep = keySteps[keyStepUpperIndex - 1] || keySteps[0];
    const upperStep = keySteps[keyStepUpperIndex] || keySteps[keySteps.length - 1];

    return [lowerStep, upperStep];
}
```
And revert changes inside this function.